client/pcr.go

```diff
@@ -81,6 +121,54 @@ func ReadPCRs(rw io.ReadWriter, sel tpm2.PCRSelection) (*pb.PCRs, error) {
 	return &pl, nil
 }
 
+// readPcrHelper fetchs the digest of a single PCR index.
+func readPcrHelper(thetpm transport.TPM, hash tpm.AlgID, index int) (*tpm2b.Digest, error) {
+	pcrRead := tpm2Direct.PCRRead{
+		PCRSelectionIn: tpml.PCRSelection{
+			PCRSelections: []tpms.PCRSelection{
+				{
+					Hash:      hash,
+					PCRSelect: make([]byte, 3),
+				},
+			},
+		},
+	}
+
+	pcrRead.PCRSelectionIn.PCRSelections[0].PCRSelect[index/8] = 1 << (index % 8)
+
+	pcrReadRsp, err := pcrRead.Execute(thetpm)
+	if err != nil {
+		return nil, fmt.Errorf("failed to pcrRead")
+	}
+
+	return &pcrReadRsp.PCRValues.Digests[0], nil
+}
+
+// ReadPCRsDirect fetches all the PCR values specified in sel, making multiple calls
+// to the TPM if necessary.
+func ReadPCRsDirect(thetpm transport.TPM, sel tpms.PCRSelection) (*pb.PCRs, error) {
+	pl := pb.PCRs{
+		Hash: pb.HashAlgo(sel.Hash),
+		Pcrs: map[uint32][]byte{},
+	}
+
+	for i, selByte := range sel.PCRSelect {
+		for j := 0; j < 8; j++ {
+			pcrIndex := i*8 + j
+			if (selByte>>j)&1 == 1 {
+				digest, err := readPcrHelper(thetpm, sel.Hash, pcrIndex)
```

Comment:
`ReadPCRs` reads in chunks of 8 PCRs. Can we do the same here to speed up this function? One option is to just call PCR_Read in the current loop with each selByte and constructing a new PCRRead command around it.

Alternatively, you can pass each byte of `sel` into readPcrHelper instead of an index.

Reply:
I talked to Chris about this as we wrote this together. We concluded that the speed up was negligible, but I will start working on the sped up version as well.

Reply:
@alexmwu would it work to have a separate PR that that just focuses on moving to the more performant version? Then we can have this PR focus on correctness?

Reply:
perfectly fine with me!